src/automata.jl

```diff
@@ -61,18 +61,18 @@ end
 
 name(s::State) = s.name
 isterminal(s::State) = s.terminal
-rightrule(s::State) = isterminal(s) ? s.rrule : nothing
+rightrule(s::State) = s.rrule
```

Comment:
Frankly, I prefer the original. There is a reason I have hidden empty word behind `nothing` when the state is terminal - the reason is that the empty word may be the right rule itself (e.g. in case aa^{-1})). Note that the job of `rightrule()`is just to get the real existing rule and the only reason we would like to access the field `s.rrule` (and not get nothing) is while declaring right rule. But this is done using `declarerightrule()` (and should always be performed using that method!)

Reply:
In the original form this method allocates (boxed value) and as such can not be used in hot loop;
revert this change and compare timings